lib/ecto/adapters/clickhouse.ex

```diff
@@ -58,9 +58,16 @@ defmodule Ecto.Adapters.ClickHouse do
       A convenience function for SQL-based repositories that translates the given query to SQL.
 
       See `Ecto.Adapters.SQL.to_sql/3` for more information.
+
+      Additional adapter-specific options include:
+
+          - `:interpolate` -- boolean for weather to interpolate params into SQL
+
       """
-      def to_sql(operation, queryable) do
-        Ecto.Adapters.ClickHouse.to_sql(operation, queryable)
+      @spec to_sql(:all | :delete_all, Ecto.Query.t(), [{:interpolate, boolean}]) ::
+              {String.t(), [term]}
+      def to_sql(operation, queryable, opts \\ []) do
+        Ecto.Adapters.ClickHouse.to_sql(operation, queryable, opts)
```

Comment:
@aerosol I'm not sure what the API should be. Should it be an option in to_sql or a completely separate function as we discussed. And what should be its name? I started with to_sql + option, but now I think we wouldn't need params to be returned (`to_sql` returns `{sql, params}`) but just the SQL, right? So now I'm leaning towards a separate function again, like `render_interpolate` or something.

Reply:
Hmm how about

```elixir
raw_sql(operation(), Ecto.Queryable.t(), Keyword.t()) :: String.t()
```

Reply:
Regardless of the name, I agree, a separate function is justified here.

Reply:
`to_sql` also produces raw sql in a sense that that's what feed ClickHouse. I went with `to_inline_sql` since I noticed that's how this procedure is [is called](https://github.com/fuelen/ecto_dev_logger/blob/da2131a7339db71b9c4c5d16e0ea50b675d988aa/lib/ecto/dev_logger.ex#L103) in Ecto.DevLogger.